src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/add-polling-source.component.spec.ts

```diff
@@ -0,0 +1,107 @@
+                ReactiveFormsModule,
+                FormsModule,
+                NgbModule,
+            ],
+            schemas: [CUSTOM_ELEMENTS_SCHEMA],
+            providers: [
+                FormBuilder,
+                Apollo,
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        snapshot: {
+                            paramMap: {
+                                get: (key: string) => {
+                                    switch (key) {
+                                        case "accountName":
+                                            return "accountName";
+                                        case "datasetName":
+                                            return "datasetName";
+                                    }
+                                },
+                            },
+                        },
+                    },
+                },
+            ],
+        })
+            .overrideComponent(AddPollingSourceComponent, {
+                set: { changeDetection: ChangeDetectionStrategy.Default },
+            })
+            .compileComponents();
+
+        fixture = TestBed.createComponent(AddPollingSourceComponent);
+        modalService = TestBed.inject(NgbModal);
+        createDatasetService = TestBed.inject(AppDatasetCreateService);
+        modalRef = modalService.open(FinalYamlModalComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+    });
+
+    it("should check open edit modal", () => {
+        component.ngOnInit();
+        const openModalSpy = spyOn(modalService, "open").and.returnValue(
+            modalRef,
+        );
+        component.onEditYaml();
+        expect(openModalSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it("should check submit yaml", () => {
+        component.ngOnInit();
+        const submitYamlSpy = spyOn(
+            createDatasetService,
+            "commitEventToDataset",
+        ).and.returnValue(new Observable());
+        component.onSubmit();
+        expect(submitYamlSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it("should check change step", () => {
+        component.ngOnInit();
+        expect(component.currentStep).toBe(SetPollingSourceSection.FETCH);
+        component.changeStep(SetPollingSourceSection.READ);
+        fixture.detectChanges();
+        expect(component.currentStep).toBe(SetPollingSourceSection.READ);
+        component.changeStep(SetPollingSourceSection.MERGE);
+        fixture.detectChanges();
+        expect(component.currentStep).toBe(SetPollingSourceSection.MERGE);
+    });
+
+    it("should check error message", () => {
+        const errorMessage = "test error message";
+        expect(component.errorMessage).toBe("");
+        createDatasetService.errorCommitEventChanges(errorMessage);
+        component.ngOnInit();
```

Comment:
Why not to move `component.ngOnInit` to the common setup, and call it always. I don't think we are testing anything useful without this method call.

Reply:
Move to the common setup.